alien-bind.js

```diff
@@ -0,0 +1,17 @@
+module.exports = alienBind
+
+function alienBind (fn, context) {
+  var boundArgs = rest(arguments, 2)
+  return function () {
+    var args = boundArgs.concat(toArray(arguments))
+    return fn.apply(context, args)
+  }
+}
+
+function rest (list, index) {
+  return Array.prototype.slice.call(list, 2)
```

Comment:
Loving the `fruit` :laughing:. I am a bit divided over whether we should change code for optimization since generally we go for readability over anything else unless it is computationally intensive. On the other hand `Array.prototype.slice` is pretty confusing and this function saves me looking up the docs every time.